src/Stryker.Core/Stryker.Core/Mutants/MutantOrchestrator.cs

```diff
@@ -62,12 +64,150 @@ public IEnumerable<Mutant> GetLatestMutantBatch()
         /// <param name="currentNode">The current root node</param>
         /// <returns>Mutated node</returns>
         public SyntaxNode Mutate(SyntaxNode currentNode)
+        {
+            // don't mutate immutable nodes
+            if (!CanBeMutated(currentNode))
+            {
+                return currentNode;
+            }
+            // apply statement specific strategies (where applicable)
+            if (currentNode is ExpressionStatementSyntax tentativeAssignment)
+            {
+                return MutateExpressionStatement(tentativeAssignment);
+            }
+            else if (currentNode is IfStatementSyntax ifStatement)
+            {
+                return MutateIfStatement(ifStatement);
+            }
+            else if (currentNode is ForStatementSyntax forStatement)
+            {
+                return MutateForStatement(forStatement);
+            }
+            // process expressions
+            var replaceNode = MutateExpressions(currentNode);
+            if (replaceNode != null)
+            {
+                return replaceNode;
+            }
+
+            // Nothing to mutate, dig further
+            var children = currentNode.ChildNodes().ToList();
+            var childCopy = currentNode.TrackNodes(children);
+            foreach (var child in children)
+            {
+                var originalNode = childCopy.GetCurrentNode(child);
+                var mutatedNode = Mutate(child);
+                if (!mutatedNode.IsEquivalentTo(originalNode))
+                {
+                    childCopy = childCopy.ReplaceNode(originalNode, mutatedNode);
+                }
+            }
+            return childCopy;
+        }
+
+        private SyntaxNode MutateExpressionStatement(ExpressionStatementSyntax expressionStatement)
+        {
+            if (expressionStatementNeedingIf.Contains(expressionStatement.Expression.GetType()))
+            {
+                var expressionCopy = expressionStatement.TrackNodes(expressionStatement, expressionStatement.Expression);
+                SyntaxNode result = MutateSubExpressionWithIfStatements(expressionStatement, expressionCopy, expressionStatement.Expression);
+                if (!(expressionStatement.Expression is AssignmentExpressionSyntax))
+                {
+                    return result;
+                }
+
+                return result.ReplaceNode(result.GetCurrentNode(expressionStatement), MutateExpressions(expressionStatement));
+            }
+
+            return MutateExpressions(expressionStatement);
+        }
+
+
+        private SyntaxNode MutateIfStatement(IfStatementSyntax ifStatement)
+        {
+            var mutatedIf = ifStatement.Else != null
+                ? ifStatement.TrackNodes(ifStatement.Condition, ifStatement.Statement, ifStatement.Else)
+                : ifStatement.TrackNodes(ifStatement.Condition, ifStatement.Statement);
+
+            if (!ifStatement.Condition.DescendantNodes().Any(x =>
```

Comment:
We could create an extensionMethod for this too like: `!ifStatement.Condition.ContainsDeclaration()`

Reply:
Done